spot-client/src/common/app-state/remote-control-service/actions.js

```diff
@@ -5,6 +5,41 @@ import { remoteControlService } from 'common/remote-control';
 import { REMOTE_CONTROL_REQUEST_STATE } from './actionTypes';
 import { requestStates, requestTypes } from './constants';
 
+/**
+ * Encapsulates updating the known status of a command in flight to a Spot-TV
+ * for a state change.
+ *
+ * @param {Function} dispatch - The Redux dispatch function to update the
+ * current state of a request in flight.
+ * @param {Function} request - The function which should be executed that
+ * performs the async request. The function must return a promise.
+ * @param {Function} requestType - The type of the request. Used to reference
+ * the request in Redux.
+ * @param {Function} expectedValue - What the expected result of the request
+ * should be. Used for optimistic updating of the UI.
+ * @private
+ * @returns {Function<Promise>}
+ */
+function createActionWithRequestStates( // eslint-disable-line max-params
+        dispatch,
+        request,
+        requestType,
+        expectedValue) {
+    dispatch(setRequestState(
+        requestType,
+        requestStates.PENDING,
+        expectedValue
+    ));
+
+    return request()
+        .then(() => dispatch(setRequestState(
+            requestType,
+            requestStates.DONE,
+            expectedValue
+        )))
+        .then(() => dispatch(setRequestState(requestType, requestStates.ERROR)));
```

Comment:
did you mean catch here ? also why not use `then(resolved, rejected)` ? doing `.catch` on `then` will also catch errors in `then` which not sure if really intended

Reply:
Oh wow, yes. catch